iib/workers/tasks/opm_operations.py

```diff
@@ -295,196 +295,132 @@ def inner(*args, **kwargs):
     return decorator
 
 
-def opm_serve_from_index(base_dir: str, from_index: str) -> Tuple[int, subprocess.Popen]:
+def get_operator_package_list(
+    input_data: str,
+    base_dir: str,
+) -> List[str]:
     """
-    Locally start OPM registry service, which can be communicated with using gRPC queries.
-
-    Due to IIB's paralellism, the service can run multiple times, which could lead to port
-    binding conflicts. Resolution of port conflicts is handled in this function as well.
+    Get list of olm.package names from input data.
 
-    :param str base_dir: base directory to create temporary files in.
-    :param str from_index: index image to inspect.
-    :return: tuple containing port number of the running service and the running Popen object.
-    :rtype: (int, Popen)
+    :param str input_data: input data for opm render
+        Example: catalog-image | catalog-directory | bundle-image | bundle-directory | sqlite-file
+    :param str base_dir: temp directory where opm will be executed.
+    :return: list of package names present in input data.
+    :rtype: [str]
     """
-    from iib.workers.tasks.build import _get_index_database
+    olm_packages = opm_render(input_data, base_dir)
 
-    log.info('Serving data from image %s', from_index)
-    if not is_image_fbc(from_index):
-        db_path = _get_index_database(from_index, base_dir)
-        return opm_registry_serve(db_path=db_path)
+    package_names = [
+        olm_package['name']
+        for olm_package in olm_packages
+        if olm_package['schema'] == 'olm.package'
+    ]
 
-    catalog_dir = get_catalog_dir(from_index, base_dir)
-    return opm_serve(catalog_dir=catalog_dir)
+    return package_names
 
 
-@create_port_filelocks(port_purposes=["opm_port", "opm_pprof_port"])
-def opm_serve(
-    opm_port: int,
-    catalog_dir: str,
-    opm_pprof_port: Optional[int] = None,
-) -> Tuple[int, subprocess.Popen]:
+def _get_olm_package_version(olm_bundle: dict) -> str:
```

Comment:
this seems more like bundle version than package version, can we rename this to_get_olm_bundle_version